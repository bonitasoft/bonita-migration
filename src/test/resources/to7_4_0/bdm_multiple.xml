<?xml version="1.0" encoding="UTF-8"?>
<def:processDefinition xmlns:def="http://www.bonitasoft.org/ns/process/client/7.2" name="Expense Investment Request Notification Accepted" version="1.2.4">
    <description/>
    <displayDescription/>
    <stringIndexes>
        <stringIndex index="1"/>
        <stringIndex index="2"/>
        <stringIndex index="3"/>
        <stringIndex index="4"/>
        <stringIndex index="5"/>
    </stringIndexes>
    <flowElements>
        <transitions>
            <transition id="7946526597618515195" name="Notify acceptance by mail_-&gt;_End of notifying acceptance" source="-7407267300811914764" target="-5358391034550328882"/>
            <transition id="4081310620509713273" name="ErrorMail_-&gt;_End of notifying acceptance" source="-4975728476977956595" target="-5358391034550328882"/>
            <transition id="4027910428827667421" name="Start Notifying accepted_-&gt;_Update acceptance" source="-7639887853639204551" target="-9061125266125452126"/>
            <transition id="802801989362862514" name="Update acceptance_-&gt;_Notify acceptance by mail" source="-9061125266125452126" target="-7407267300811914764"/>
        </transitions>
        <connectors/>
        <businessDataDefinitions>
            <businessDataDefinition className="com.company.bdm.Creator" name="creator">
                <description/>
            </businessDataDefinition>
            <businessDataDefinition className="com.company.bdm.ExpenseInvestmentRequest" name="expenseInvestmentRequest">
                <description/>
            </businessDataDefinition>
        </businessDataDefinitions>
        <dataDefinitions/>
        <documentDefinitions/>
        <automaticTask id="-9061125266125452126" name="Update acceptance">
            <incomingTransition>4027910428827667421</incomingTransition>
            <outgoingTransition>802801989362862514</outgoingTransition>
            <connector activationEvent="ON_ENTER" connectorId="database-datasource" failAction="FAIL" name="updateProposalAccepted" version="1.0.0">
                <inputs>
                    <input name="properties">
                        <expression expressionType="TYPE_LIST" name="Table of expression containing the following expressions: []." returnType="java.util.List" id="4789497637205100328">
                            <content>Table of expression containing the following expressions: [].</content>
                        </expression>
                    </input>
                    <input name="script">
                        <expression expressionType="TYPE_READ_ONLY_SCRIPT" interpreter="GROOVY" name="getUpdateProposalAcceptedQuery" returnType="java.lang.String" id="9107289489407504969">
                            <content>return "*"</content>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="expenseInvestmentRequest" returnType="com.company.bdm.ExpenseInvestmentRequest" id="5481903788488236833">
                                <content>expenseInvestmentRequest</content>
                            </expression>
                        </expression>
                    </input>
                    <input name="dataSourceName">
                        <expression expressionType="TYPE_PARAMETER" interpreter="" name="dbDatasource" returnType="java.lang.String" id="7788781773947893830">
                            <content>dbDatasource</content>
                        </expression>
                    </input>
                </inputs>
                <outputs/>
            </connector>
            <dataDefinitions/>
            <operations/>
            <boundaryEvents/>
        </automaticTask>
        <automaticTask id="-7407267300811914764" name="Notify acceptance by mail">
            <incomingTransition>802801989362862514</incomingTransition>
            <outgoingTransition>7946526597618515195</outgoingTransition>
            <connector activationEvent="ON_ENTER" connectorId="email" errorCode="ErrorMailOnNotificationAccepted" failAction="ERROR_EVENT" name="coMail_AcceptNotification" version="1.0.0">
                <inputs>
                    <input name="headers">
                        <expression expressionType="TYPE_LIST" name="Table of expression containing the following expressions: []." returnType="java.util.List" id="7114557732345107763">
                            <content>Table of expression containing the following expressions: [].</content>
                        </expression>
                    </input>
                    <input name="message">
                        <expression expressionType="TYPE_READ_ONLY_SCRIPT" interpreter="GROOVY" name="getMessage" returnType="java.lang.String" id="7277600823004641191">
                            <content>StringBuffer buffer = new StringBuffer();
                                return buffer.toString();</content>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="expenseInvestmentRequest" returnType="com.company.bdm.ExpenseInvestmentRequest" id="8725487515056518087">
                                <content>expenseInvestmentRequest</content>
                            </expression>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="creator" returnType="com.company.bdm.Creator" id="7108518638169660070">
                                <content>creator</content>
                            </expression>
                        </expression>
                    </input>
                    <input name="to">
                        <expression expressionType="TYPE_READ_ONLY_SCRIPT" interpreter="GROOVY" name="getCreatorEmail" returnType="java.lang.String" id="6510409137950339381">
                            <content>return creator.getEmail();</content>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="creator" returnType="com.company.bdm.Creator" id="5166037581930912549">
                                <content>creator</content>
                            </expression>
                        </expression>
                    </input>
                    <input name="smtpPort">
                        <expression expressionType="TYPE_PARAMETER" interpreter="" name="mailPort" returnType="java.lang.Integer" id="8265449152599125202">
                            <content>mailPort</content>
                        </expression>
                    </input>
                    <input name="starttlsSupport">
                        <expression expressionType="TYPE_CONSTANT" interpreter="" name="false" returnType="java.lang.Boolean" id="6036115650361030290">
                            <content>false</content>
                        </expression>
                    </input>
                    <input name="sslSupport">
                        <expression expressionType="TYPE_CONSTANT" interpreter="" name="false" returnType="java.lang.Boolean" id="6765766991685847771">
                            <content>false</content>
                        </expression>
                    </input>
                    <input name="smtpHost">
                        <expression expressionType="TYPE_PARAMETER" interpreter="" name="mailSMTPHost" returnType="java.lang.String" id="5695217238088369485">
                            <content>mailSMTPHost</content>
                        </expression>
                    </input>
                    <input name="subject">
                        <expression expressionType="TYPE_READ_ONLY_SCRIPT" interpreter="GROOVY" name="getSubject" returnType="java.lang.String" id="6040050300966265233">
                            <content>return "["+expenseInvestmentRequest.getTypeOfRequestLabel()+"] Your "+expenseInvestmentRequest.getTypeOfRequestLabel()+" "+expenseInvestmentRequest.getNumber()+" has been accepted"</content>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="expenseInvestmentRequest" returnType="com.company.bdm.ExpenseInvestmentRequest" id="6224690210554723418">
                                <content>expenseInvestmentRequest</content>
                            </expression>
                        </expression>
                    </input>
                    <input name="charset">
                        <expression expressionType="TYPE_CONSTANT" interpreter="" name="UTF-8" returnType="java.lang.String" id="6965147544565070783">
                            <content>UTF-8</content>
                        </expression>
                    </input>
                    <input name="html">
                        <expression expressionType="TYPE_CONSTANT" interpreter="" name="true" returnType="java.lang.Boolean" id="7246830384223023450">
                            <content>true</content>
                        </expression>
                    </input>
                    <input name="attachments">
                        <expression expressionType="TYPE_LIST" name="List of expression containing the following expressions: ()." returnType="java.util.List" id="5458585534291703522">
                            <content>List of expression containing the following expressions: ().</content>
                        </expression>
                    </input>
                    <input name="from">
                        <expression expressionType="TYPE_PARAMETER" interpreter="" name="mailSender" returnType="java.lang.String" id="8049037882104144593">
                            <content>mailSender</content>
                        </expression>
                    </input>
                </inputs>
                <outputs/>
            </connector>
            <dataDefinitions/>
            <operations/>
            <boundaryEvents>
                <boundaryEvent id="-4975728476977956595" interrupting="true" name="ErrorMail">
                    <outgoingTransition>4081310620509713273</outgoingTransition>
                    <catchErrorEventTrigger errorCode="ErrorMailOnNotificationAccepted"/>
                </boundaryEvent>
            </boundaryEvents>
        </automaticTask>
        <startEvent id="-7639887853639204551" interrupting="true" name="Start Notifying accepted">
            <outgoingTransition>4027910428827667421</outgoingTransition>
        </startEvent>
        <endEvent id="-5358391034550328882" name="End of notifying acceptance">
            <incomingTransition>7946526597618515195</incomingTransition>
            <incomingTransition>4081310620509713273</incomingTransition>
            <terminateEventTrigger/>
        </endEvent>
    </flowElements>
    <parameters>
        <parameter name="mailPassword" type="java.lang.String">
            <description/>
        </parameter>
        <parameter name="mailSMTPHost" type="java.lang.String">
            <description/>
        </parameter>
        <parameter name="mailPort" type="java.lang.Integer">
            <description/>
        </parameter>
        <parameter name="mailSender" type="java.lang.String">
            <description/>
        </parameter>
        <parameter name="dbDatasource" type="java.lang.String">
            <description/>
        </parameter>
        <parameter name="mailUsername" type="java.lang.String">
            <description/>
        </parameter>
    </parameters>
    <actors/>
</def:processDefinition>
