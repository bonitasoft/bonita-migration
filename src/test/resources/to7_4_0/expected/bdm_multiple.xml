<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<def:processDefinition xmlns:def="http://www.bonitasoft.org/ns/process/client/7.4" name="Expense Investment Request Notification Accepted" version="1.2.4">
    <description/>
    <displayDescription/>
    <parameters>
        <parameter name="mailPassword" type="java.lang.String">
            <description/>
        </parameter>
        <parameter name="mailSMTPHost" type="java.lang.String">
            <description/>
        </parameter>
        <parameter name="mailPort" type="java.lang.Integer">
            <description/>
        </parameter>
        <parameter name="mailSender" type="java.lang.String">
            <description/>
        </parameter>
        <parameter name="dbDatasource" type="java.lang.String">
            <description/>
        </parameter>
        <parameter name="mailUsername" type="java.lang.String">
            <description/>
        </parameter>
    </parameters>
    <actors/>
    <flowElements>
        <automaticTask id="_-9061125266125452126" name="Update acceptance">
            <incomingTransition>_4027910428827667421</incomingTransition>
            <outgoingTransition>_802801989362862514</outgoingTransition>
            <connector activationEvent="ON_ENTER" connectorId="database-datasource" failAction="FAIL" name="updateProposalAccepted" version="1.0.0">
                <inputs>
                    <input name="properties">
                        <expression expressionType="TYPE_LIST" name="Table of expression containing the following expressions: []." returnType="java.util.List" id="_65609">
                            <content>Table of expression containing the following expressions: [].</content>
                        </expression>
                    </input>
                    <input name="script">
                        <expression expressionType="TYPE_READ_ONLY_SCRIPT" interpreter="GROOVY" name="getUpdateProposalAcceptedQuery" returnType="java.lang.String" id="_65619">
                            <content>return "*"</content>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="expenseInvestmentRequest" returnType="com.company.bdm.ExpenseInvestmentRequest" id="_65626">
                                <content>expenseInvestmentRequest</content>
                            </expression>
                        </expression>
                    </input>
                    <input name="dataSourceName">
                        <expression expressionType="TYPE_PARAMETER" interpreter="" name="dbDatasource" returnType="java.lang.String" id="_65636">
                            <content>dbDatasource</content>
                        </expression>
                    </input>
                </inputs>
                <outputs/>
            </connector>
            <dataDefinitions/>
            <operations/>
            <boundaryEvents/>
        </automaticTask>
        <automaticTask id="_-7407267300811914764" name="Notify acceptance by mail">
            <incomingTransition>_802801989362862514</incomingTransition>
            <outgoingTransition>_7946526597618515195</outgoingTransition>
            <connector activationEvent="ON_ENTER" connectorId="email" errorCode="ErrorMailOnNotificationAccepted" failAction="ERROR_EVENT" name="coMail_AcceptNotification" version="1.0.0">
                <inputs>
                    <input name="headers">
                        <expression expressionType="TYPE_LIST" name="Table of expression containing the following expressions: []." returnType="java.util.List" id="_65664">
                            <content>Table of expression containing the following expressions: [].</content>
                        </expression>
                    </input>
                    <input name="message">
                        <expression expressionType="TYPE_READ_ONLY_SCRIPT" interpreter="GROOVY" name="getMessage" returnType="java.lang.String" id="_65674">
                            <content>StringBuffer buffer = new StringBuffer();
                                return buffer.toString();</content>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="expenseInvestmentRequest" returnType="com.company.bdm.ExpenseInvestmentRequest" id="_65681">
                                <content>expenseInvestmentRequest</content>
                            </expression>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="creator" returnType="com.company.bdm.Creator" id="_65688">
                                <content>creator</content>
                            </expression>
                        </expression>
                    </input>
                    <input name="to">
                        <expression expressionType="TYPE_READ_ONLY_SCRIPT" interpreter="GROOVY" name="getCreatorEmail" returnType="java.lang.String" id="_65698">
                            <content>return creator.getEmail();</content>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="creator" returnType="com.company.bdm.Creator" id="_65705">
                                <content>creator</content>
                            </expression>
                        </expression>
                    </input>
                    <input name="smtpPort">
                        <expression expressionType="TYPE_PARAMETER" interpreter="" name="mailPort" returnType="java.lang.Integer" id="_65715">
                            <content>mailPort</content>
                        </expression>
                    </input>
                    <input name="starttlsSupport">
                        <expression expressionType="TYPE_CONSTANT" interpreter="" name="false" returnType="java.lang.Boolean" id="_65725">
                            <content>false</content>
                        </expression>
                    </input>
                    <input name="sslSupport">
                        <expression expressionType="TYPE_CONSTANT" interpreter="" name="false" returnType="java.lang.Boolean" id="_65735">
                            <content>false</content>
                        </expression>
                    </input>
                    <input name="smtpHost">
                        <expression expressionType="TYPE_PARAMETER" interpreter="" name="mailSMTPHost" returnType="java.lang.String" id="_65745">
                            <content>mailSMTPHost</content>
                        </expression>
                    </input>
                    <input name="subject">
                        <expression expressionType="TYPE_READ_ONLY_SCRIPT" interpreter="GROOVY" name="getSubject" returnType="java.lang.String" id="_65755">
                            <content>return "["+expenseInvestmentRequest.getTypeOfRequestLabel()+"] Your "+expenseInvestmentRequest.getTypeOfRequestLabel()+" "+expenseInvestmentRequest.getNumber()+" has been accepted"</content>
                            <expression expressionType="TYPE_BUSINESS_DATA" name="expenseInvestmentRequest" returnType="com.company.bdm.ExpenseInvestmentRequest" id="_65762">
                                <content>expenseInvestmentRequest</content>
                            </expression>
                        </expression>
                    </input>
                    <input name="charset">
                        <expression expressionType="TYPE_CONSTANT" interpreter="" name="UTF-8" returnType="java.lang.String" id="_65772">
                            <content>UTF-8</content>
                        </expression>
                    </input>
                    <input name="html">
                        <expression expressionType="TYPE_CONSTANT" interpreter="" name="true" returnType="java.lang.Boolean" id="_65782">
                            <content>true</content>
                        </expression>
                    </input>
                    <input name="attachments">
                        <expression expressionType="TYPE_LIST" name="List of expression containing the following expressions: ()." returnType="java.util.List" id="_65791">
                            <content>List of expression containing the following expressions: ().</content>
                        </expression>
                    </input>
                    <input name="from">
                        <expression expressionType="TYPE_PARAMETER" interpreter="" name="mailSender" returnType="java.lang.String" id="_65801">
                            <content>mailSender</content>
                        </expression>
                    </input>
                </inputs>
                <outputs/>
            </connector>
            <dataDefinitions/>
            <operations/>
            <boundaryEvents>
                <boundaryEvent id="_-4975728476977956595" interrupting="true" name="ErrorMail">
                    <outgoingTransition>_4081310620509713273</outgoingTransition>
                    <catchErrorEventTrigger errorCode="ErrorMailOnNotificationAccepted"/>
                </boundaryEvent>
            </boundaryEvents>
        </automaticTask>
        <transitions>
            <transition id="_7946526597618515195" name="Notify acceptance by mail_-&gt;_End of notifying acceptance" source="_-7407267300811914764" target="_-5358391034550328882"/>
            <transition id="_4081310620509713273" name="ErrorMail_-&gt;_End of notifying acceptance" source="_-4975728476977956595" target="_-5358391034550328882"/>
            <transition id="_4027910428827667421" name="Start Notifying accepted_-&gt;_Update acceptance" source="_-7639887853639204551" target="_-9061125266125452126"/>
            <transition id="_802801989362862514" name="Update acceptance_-&gt;_Notify acceptance by mail" source="_-9061125266125452126" target="_-7407267300811914764"/>
        </transitions>
        <startEvent id="_-7639887853639204551" interrupting="true" name="Start Notifying accepted">
            <outgoingTransition>_4027910428827667421</outgoingTransition>
        </startEvent>
        <endEvent id="_-5358391034550328882" name="End of notifying acceptance">
            <incomingTransition>_7946526597618515195</incomingTransition>
            <incomingTransition>_4081310620509713273</incomingTransition>
            <terminateEventTrigger/>
        </endEvent>
        <dataDefinitions/>
        <businessDataDefinitions>
            <businessDataDefinition className="com.company.bdm.Creator" name="creator" multiple="false">
                <description/>
            </businessDataDefinition>
            <businessDataDefinition className="com.company.bdm.ExpenseInvestmentRequest" name="expenseInvestmentRequest" multiple="false">
                <description/>
            </businessDataDefinition>
        </businessDataDefinitions>
        <documentDefinitions/>
        <connectors/>
    </flowElements>
    <stringIndexes>
        <stringIndex index="1"/>
        <stringIndex index="2"/>
        <stringIndex index="3"/>
        <stringIndex index="4"/>
        <stringIndex index="5"/>
    </stringIndexes>
</def:processDefinition>